python/tests/test_file_format.py

```diff
@@ -779,3 +780,20 @@ def test_load_bad_formats(self):
         with open(self.temp_file, "wb") as f:
             f.write(os.urandom(8192))
         self.assertRaises(exceptions.FileFormatError, tskit.load, self.temp_file)
+
+
+class TestPickle(TestFileFormat):
+    """
+    Test pickling of a TreeSequence.
+    """
```

Comment:
This is great! To generalise a little bit, I'd do this:
```python
def verify_round_trip(self, ts):
    pkl = pickle.dumps(ts)
    tsp = pickle.loads(pkl)
    self.assertEqual(ts.tables, tsp.tables)

def test_simple(self):
    self.verify_round_trip(msprime.simulate(10, random_seed=2))

def test_recombination(self):
    self.verify_round_trip(msprime.simulate(10, recombation_rate=1, random_seed=2))

def test_mutations(self):
    self.verify_round_trip(msprime.simulate(10, mutation_rate=1, random_seed=2))

def test_migrations(self):
    ts = msprime.simulate(
          population_configurations=[
              msprime.PopulationConfiguration(10),
              msprime.PopulationConfiguration(10)],
          migration_matrix=[[1,0], [0, 1]],
          record_migrations=True,
          random_seed=2)
         self.verify_round_trip(ts)
```
That basically catches all the important stuff.